### PR TITLE
Add parameter to get tasks in reverse order

### LINKED ISSUE #693: [v1.12.0] `reverse` parameter when querying tasks
Related to https://github.com/meilisearch/integration-guides/issues/307

### Explanation of the feature

Add a new parameter in `GET /tasks` route: `reverse`

Usage: https://meilisearch.notion.site/Usage-Reverse-the-order-of-the-task-queue-1364b06b651f81dcb618e8409d30429b

### TODO

- [ ] Add a new query parameter associated to the `getTasks` method
- [ ] Add tests

⚠️ Make PRs pointing to `bump-meilisearch-v1.12.0` and NOT `main`. Please do 1 PR for all of these changes, and not several.

### PR DESCRIPTION
**Pull request by Strift**
# Pull Request

## Related issue
Fixes #693

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: src/Contracts/TasksQuery.php [hunk 1]
```diff
@@ -63,6 +75,7 @@ public function toArray(): array
                     'from' => $this->from,
                     'limit' => $this->limit,
                     'canceledBy' => $this->formatArray($this->canceledBy),
+                    'reverse' => null !== $this->reverse ? json_encode($this->reverse) : null,
```
Comment (norkunas) [inline src/Contracts/TasksQuery.php:78]: https://github.com/meilisearch/meilisearch-php/blob/ef1769975bcc3b97057dc6ee3ed3f733a8ec7f0e/src/Contracts/DocumentsQuery.php#L119

avoid json_encode to be consistent :)

File: tests/Contracts/TasksQueryTest.php [hunk 2]
```diff
@@ -109,4 +109,11 @@ public function testSetFrom(): void
 
         self::assertSame(['from' => 1], $data->toArray());
     }
+
+    public function testSetReverse(): void
+    {
+        $data = (new TasksQuery())->setReverse(true);
+
+        self::assertSame(['reverse' => json_encode(true)], $data->toArray());
```
Comment (norkunas) [inline tests/Contracts/TasksQueryTest.php:117]:
```suggestion
        self::assertSame(['reverse' => 'true'], $data->toArray());
```

File: tests/Endpoints/TasksTest.php [hunk 3]
```diff
@@ -140,6 +140,19 @@ public function testCancelTasksWithFilter(): void
         self::assertSame('succeeded', $response['status']);
     }
 
+    public function testGetAllTasksInReverseOrder(): void
+    {
+        $date = new \DateTime();
```
Comment (norkunas) [inline tests/Endpoints/TasksTest.php:145]:
```suggestion
        $date = new \DateTimeImmutable();
```

File: tests/Contracts/TasksQueryTest.php [hunk 4]
```diff
@@ -109,4 +109,11 @@ public function testSetFrom(): void
...
+    }
```
Comment (norkunas) [inline tests/Contracts/TasksQueryTest.php:118]:
```suggestion
    /**
     * @testWith [true, "true"]
     *                     [false, "false"]
     */
    public function testSetReverse(bool $reverse, string $expected): void
    {
        $data = (new TasksQuery())->setReverse($reverse);

        self::assertSame(['reverse' => $expected], $data->toArray());
    }
```

File: src/Contracts/TasksQuery.php (hunk 1 above)
Comment (Strift) [reply src/Contracts/TasksQuery.php]: Thank you for the suggestion!

**Comment by Strift** [comment]
Hello @norkunas and thanks for the early review 👋  I forgot to set the PR as a draft.

I applied all your suggestions. Thanks! We're now ready for a new review. 

It looks like the test is flaky though, because it failed at first; and worked when I re-ran it. I'm not sure how I can make it better. If you have feedback, it's welcome.

**Comment by norkunas** [comment]
> Hello @norkunas and thanks for the early review 👋 I forgot to set the PR as a draft.

Hey :wave: 

> It looks like the test is flaky though, because it failed at first; and worked when I re-ran it. I'm not sure how I can make it better.

Which tasks does it retrieve? Because I don't see that index would be seeded?

**Comment by norkunas** [comment]
Oh looks like that after each test the index is not cleared, so it sees previous tests data :expressionless:

**Review by norkunas — APPROVED** [review]
> Would you recommend updating the tests or do you think it's sufficient as is?

I'd at least add a (todo) note that these tests should be refactored to cleanup after themself

**Pull request merged**